service_container/3.3-di-changes.rst

```diff
@@ -0,0 +1,740 @@
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <defaults autowire="true" autoconfigure="true" public="false" />
+
+                <prototype namespace="AppBundle\" resource="../../src/AppBundle/*" exclude="../../src/AppBundle/{Entity,Repository}" />
+
+                <prototype namespace="AppBundle\Controller" resource="../../src/AppBundle/Controller" public="true">
+                    <tag name="controller.service_arguments" />
+                </prototype>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+
+        // _defaults and loading entire directories is not possible with PHP configuration
+        // you need to define your services one-by-one
+        use AppBundle\Controller\DefaultController;
+
+        $container->autowire(DefaultController::class)
+            ->setAutoconfigured(true)
+            ->addTag('controller.service_arguments')
+            ->setPublic(false);
+
+This small bit of configuration contains some big philosophical changes to how services
+are configured in Symfony.
+
+.. _`service-33-changes-automatic-registration`:
+
+1) Services are Loaded Automatically
+------------------------------------
+
+.. seealso::
+
+    Read the documentation for :ref:`automatic service loading <service-psr4-loader>`.
+
+The first big change is that services do *not* need to be defined one-by-one anymore,
+thanks to the following config:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            # ...
+
+            # makes classes in src/AppBundle available to be used as services
+            # this creates a service per class whose id is the fully-qualified class name
+            AppBundle\:
+                resource: '../../src/AppBundle/*'
+                # you can exclude directories or files
+                # but if a service is unused, it's removed anyway
+                exclude: '../../src/AppBundle/{Entity,Repository}'
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <prototype namespace="AppBundle\" resource="../../src/AppBundle/*" exclude="../../src/AppBundle/{Entity,Repository}" />
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+
+        // services cannot be automatically loaded with PHP configuration
```

Comment:
Of course! I didn't realize that - I've opened an issue to update this everywhere: https://github.com/symfony/symfony-docs/issues/7922